
<launch>
    <param name="use_sim_time" value="true"/>
	<!-- tf -->
    <include file="$(find cats)/launch/robot_state.launch" />
    <include file="$(find making_local_map)/launch/camera_tf.launch" />

	<!-- velodyne -->
    <include file="$(find velodyne_pointcloud)/launch/32e_points.launch" />

	<!-- downsampling realsense points -->
	<node pkg="making_local_map" type="downsample_and_transform" name="downsample_and_transform">
		<remap from="/cloud" to="/camera/depth_registered/points"/>
		<remap from="/cloud/downsampled" to="/realsense/downsampled"/>
	</node>

	<!-- min_max -->
    <include file="$(find velodyne_height_map)/launch/amsl_velodyne_heightmap.launch" />
	<!-- <node pkg="nodelet" type="nodelet" name="heightmap_realsense" -->
    <!--     args="load velodyne_height_map/HeightMapNodelet velodyne_nodelet_manager"> -->
	<!-- 	<param name="cell_size" value="0.2"/> -->
	<!-- 	<param name="full_clouds" value="true"/> -->
	<!-- 	<param name="grid_dimensions" value="25"/> -->
	<!-- 	<param name="height_threshold" value="0.2"/> -->
	<!-- 	<remap from="/velodyne_points" to="/realsense/downsampled"/> -->
	<!-- 	<remap from="/velodyne_obstacles" to="/realsense/obstacles"/> -->
	<!-- 	<remap from="/velodyne_clear" to="/realsense/clear"/> -->
	<!-- </node> -->

	<node pkg="making_local_map" type="realsense_object" name="realsense_obstacle" output="screen">
		<remap from="/cloud" to="/realsense/downsampled"/>
		<remap from="/cloud/curv" to="/realsense/obstacles"/>
		<remap from="/cloud/plane" to="/realsense/clear"/>
		<param name="UPPER_LIMIT" value="0.15" type="double"/>
		<param name="UNDER_LIMIT" value="-0.1" type="double"/>
		<param name="VIEW_RANGE_X_MIN" value="0.3" type="double"/>
		<param name="VIEW_RANGE_X_MAX" value="1.5" type="double"/>
		<param name="VIEW_RANGE_Y_MIN" value="-0.8" type="double"/>
		<param name="VIEW_RANGE_Y_MAX" value="0.8" type="double"/>
		<param name="VIEW_RANGE_Z_MIN" value="-0.5" type="double"/>
		<param name="VIEW_RANGE_Z_MAX" value="0.5" type="double"/>
	</node>

	<!-- hokuyo -->
	<node pkg="making_local_map" type="scan2pcl" name="scan2pcl" output="screen">
		<remap from="/scan" to="/front_lasr/scan"/>
		<remap from="/cloud" to="/hokuyo_obstacles"/>
	</node>


	<!-- make each occupancygrid -->
		<!-- velodyne -->
	<node pkg="making_local_map" type="ground_and_obstacles_to_grid" name="velodyne_localmap" output="screen">
		<remap from="/rm_ground" to="/velodyne_obstacles"/>
		<remap from="/ground" to="/velodyne_clear"/>
		<remap from="/occupancygrid/lidar" to="/occupancygrid/velodyne"/>
		<remap from="/grass_points" to="/grass_points"/>
		<param name="ROAD_INTENSITY_MIN" value="1" type="double"/>
		<param name="ROAD_INTENSITY_MAX" value="15" type="double"/>
		<param name="ZEROCELL_RATIO" value="0.4" type="double"/>
		<param name="FILTER_RANGE" value="3" type="int"/>
	</node>
		<!-- realsense -->
	<node pkg="making_local_map" type="realsense_to_grid" name="realsense_localmap">
		<remap from="/rm_ground" to="/realsense/obstacles"/>
		<remap from="/ground" to="/realsense/clear"/>
		<remap from="/occupancygrid/lidar" to="/occupancygrid/realsense"/>
	</node>
		<!-- hokuyo -->
	<node pkg="making_local_map" type="hokuyo_raycast" name="hokuyo_localmap">
		<remap from="/hokuyo_points" to="/hokuyo_obstacles"/>
		<remap from="/occupancygrid/lidar" to="/occupancygrid/hokuyo"/>
	</node>

	<!-- integrate localmap -->
	<node pkg="making_local_map" type="occupancygrid_integrate" name="occupancygrid_integrate">
		<remap from="/occupancygrid/lidar/stored" to="/occupancygrid/velodyne"/>
		<remap from="/occupancygrid/realsense" to="/occupancygrid/realsense"/>
		<remap from="/occupancygrid/hokuyo" to="/occupancygrid/hokuyo"/>
		<param name="EXPAND_RANGE" value="3" type="int"/>
		<param name="EXPAND_RANGE_MINI" value="1" type="int"/>
	</node>

	<!-- rviz -->
	<node pkg="rviz" type="rviz" name="rviz" args="/home/amsl/rviz/test.rviz"/>

	<!-- rosbag play -->
	<!-- <node pkg="rosbag" type="play" name="play" args="&#45;&#45;clock /home/amsl/bagfiles/realsense_2019&#45;05&#45;16&#45;16&#45;41&#45;13.bag &#45;l" /> -->
	<node pkg="rosbag" type="play" name="play" args="--clock /home/amsl/bagfiles/2018-09-15-16-11-46.bag -s 300 -l" />
	<!-- <node pkg="rosbag" type="play" name="play" args="&#45;&#45;clock /home/amsl/bagfiles/2018&#45;06&#45;30&#45;15&#45;29&#45;21.bag &#45;s 150 &#45;l" /> -->
</launch>
